### ISSUE TITLE
Lost files in packages

### ISSUE THREAD

**Issue by gueniewie** [ladar/qemu-spice-el9#3]
Hello,
the qemu-kvm and qemu-kvm-core have 0 files in the rpm Packages and this are missing by installing create and start a KVM?

PLEASE can you have a look for a correct build of the Pakage

Thanks for your work,
Günther

**Comment by agibson2** [comment]
Have you looked at the other fork of this at the github link below?  It appears that project don't host any RPMs but instead they have a build script that downloads the source RPMs (latest) directly from fedora source and only patch and recompile the ones that are necessary.
https://github.com/ambakshi/qemu-spice-el9

**Comment by ladar** [comment]
Yeah, I think the other variant uses the RHEL versions whereas I recompile the Fedora release for RHEL.

**Comment by gueniewie** [comment]
Thank you for your work, I hope this Build will work ;.) and the newer rpm's can be integrat.

Again thank's for your help and your time

**Comment by agibson2** [comment]
Yea the other variant seems to only download  the packages from fedora that aren't in RHEL 9 and applies patches to what needs them.  The rest of them they pull from RHEL packages.  The funny part though is that because that variant is a clone of this repo, The git repo when you clone that variant, is pretty large as it contains all the packages from this repo but they are removed in the tree.

You spawned all this work though it seems so I thank you very much.  We have options now.

**Comment by ladar** [comment]
@gueniewie can you try the new set RPMs / install script? It includes a rebuilt `qemu-kvm` RPM package. It also sets up a symlink between `/usr/libexec/qemu-kvm` and the standardized `qemu-system-x86_64` binary.

**Comment by ladar** [comment]
> The git repo when you clone that variant, is pretty large as it contains all the packages from this repo but they are removed in the tree.

To avoid the massive download, use the '--depth=1' option:

```bash
# git clone --depth=1 https://github.com/ambakshi/qemu-spice-el9
Cloning into 'qemu-spice-el9'...
remote: Enumerating objects: 8, done.
remote: Counting objects: 100% (8/8), done.
remote: Compressing objects: 100% (8/8), done.
remote: Total 8 (delta 0), reused 4 (delta 0), pack-reused 0
Receiving objects: 100% (8/8), done.
# du -shc qemu-spice-el9/
148K	qemu-spice-el9/
148K	total
```

At some point soon, it will make sense to make shallow clones for this this repo as well. The more I rebuild the RPMs, the larger the history will get. If I get some time I'll add a note to the README suggesting people simply grab the archive and/or if they use git, to make a shallow clone.

**Comment by gueniewie** [comment]
#git clone --depth=1 https://github.com/ambakshi/qemu-spice-el9

This clone is broken ? But I set up a new Alma Linux 9.1 in a Virtualbox to Test your INSTALL.sh

Is it also possible in your git Brach to set the --depth 1 command ? the download is very big ;-)

**Comment by gueniewie** [comment]
Hello Ladar,

The INSTALL.sh script is in the MOMENT broken, the script can not find the correct Pakages ? I have allways this Errors I mean with all Packages ?

I mean I have to find out the LATEST Packages and delete the older, to install it ?


 Problem 1: cannot install both edk2-aarch64-20221117gitfff6d81270b5-12.btrh9.noarch and edk2-aarch64-20221117gitfff6d81270b5-10.btrh9.noarch
  - widersprüchliche Anforderungen
 Problem 2: cannot install both edk2-arm-20221117gitfff6d81270b5-12.btrh9.noarch and edk2-arm-20221117gitfff6d81270b5-10.btrh9.noarch
  - widersprüchliche Anforderungen
 Problem 3: cannot install both edk2-ovmf-20221117gitfff6d81270b5-12.btrh9.noarch and edk2-ovmf-20221117gitfff6d81270b5-10.btrh9.noarch
...........
 Problem 127: cannot install both virglrenderer-0.10.4-2.20230104git88b9fe3b.btrh9.x86_64 and virglrenderer-0.10.4-1.20230104git88b9fe3b.btrh9.x86_64
  - widersprüchliche Anforderungen
 Problem 128: cannot install both virglrenderer-devel-0.10.4-2.20230104git88b9fe3b.btrh9.x86_64 and virglrenderer-devel-0.10.4-1.20230104git88b9fe3b.btrh9.x86_64
  - widersprüchliche Anforderungen
 Problem 129: cannot install both virglrenderer-test-server-0.10.4-2.20230104git88b9fe3b.btrh9.x86_64 and virglrenderer-test-server-0.10.4-1.20230104git88b9fe3b.btrh9.x86_64
  - widersprüchliche Anforderungen
 Problem 130: cannot install both virt-install-4.1.0-2.btrh9.noarch and virt-install-4.1.0-1.btrh9.noarch
  - widersprüchliche Anforderungen
 Problem 131: cannot install both virt-manager-4.1.0-2.btrh9.noarch and virt-manager-4.1.0-1.btrh9.noarch
  - widersprüchliche Anforderungen
 Problem 132: cannot install both virt-manager-common-4.1.0-2.btrh9.noarch and virt-manager-common-4.1.0-1.btrh9.noarch
  - widersprüchliche Anforderungen
Nichts zu tun.

**Comment by ladar** [comment]
> #git clone --depth=1 https://github.com/ambakshi/qemu-spice-el9
> 
> This clone is broken ? But I set up a new Alma Linux 9.1 in a Virtualbox to Test your INSTALL.sh
> 
> Is it also possible in your git Brach to set the --depth 1 command ? the download is very big ;-)

That command was to clone the forked version of the repo, but without the now removed RPM files.

**Comment by ladar** [comment]
> cannot install both edk2-aarch64-20221117gitfff6d81270b5-12.btrh9.noarch and edk2-aarch64-20221117gitfff6d81270b5-10.btrh9.noarch

Looks like the system you were trying to install on already had an older version the edk2 package installed. I'm not sure why it isn't simply upgrading the package. It's possible the commit string in the package name is breaking the version comparison. 

A potential would be to edit [this](https://github.com/ladar/qemu-spice-el9/blob/da0a3382c33101123690be72351b8ed6c54bbd95/INSTALL.sh#L21) line and [this](https://github.com/ladar/qemu-spice-el9/blob/da0a3382c33101123690be72351b8ed6c54bbd95/INSTALL.sh#L23) line in the `INSTALL.sh` script. Change `install` to `reinstall` ...

**Comment by ladar** [comment]
Actually @gueniewie I'd like you to test a different strategy. Can you try using the attached version of the `INSTALL.sh` script instead? I added the edk2 packages to the remove list. This way, at least in theory, the script will detect and remove them (if they are already installed) before installing the new set of replacement packages. It's all supposed to happen as a transaction to avoid issues with any dependencies that might be installed.

The downside of this approach is that I think this logic will always cause the edk2 packages to get reinstalled. But it should workaround the issue you ran into. and be a better long term strategy. If it works, I'll update the repo.

[INSTALL.sh.txt](https://github.com/ladar/qemu-spice-el9/files/10755226/INSTALL.sh.txt)

**Comment by ladar** [comment]
Acutally, give me another minute. After looking more closely at your Spanish error messages, I realized I need to make another change.

**Comment by gueniewie** [comment]
OK, I change this, but this was a clear NEW Installation "Server with GUI"?
In the Moment I test the BUILD.sh but I mean I have a Problem with the elrepo-release-9.1-1 its waiting now ~1h I mean, I make a New start ;-)

**Comment by ladar** [comment]
This version should handle the virglrenderer and virt-manager/virt-install conflicts.

[INSTALL.sh.txt](https://github.com/ladar/qemu-spice-el9/files/10755393/INSTALL.sh.txt)

**Comment by ladar** [comment]
The "Server with GUI" profile probably installs QEMU/libvirt/virt-manager, during the installation. The `INSTALL.sh` script was designed to detect distro versions of the packages, which might be installed already, and remove them during the installation process.

But when I originally wrote the script, I wasn't rebuilding virglrenderer and virt-manager/virt-install, so I didn't include them in the list.

In theory you can always remove the conflicting packages manually before running `INSTALL.sh` ... but that would normally also remove any dependent packages. Hence the need to handle them as part of the transaction.

**Comment by gueniewie** [comment]
Thanks for your work, but I am not a Programmer, so I must waiting for you :-)

OK, I am not so included with RH or other Linuxes, but in the moment I test the BUILD.sh and looking the ~600-?? Packets that was installed for building the new Packets when it is OK, I like to insert this in a repository LOCAL.

I have to update my 2 small Server to alma 9 or oracle 9 ??

**Comment by ladar** [comment]
I run `BUILD.sh` inside a disposable virtual machine. As a result, I used package groups to grab and install as many development/programming packages as possible, It would have taken a lot longer to idendify just the specific packages needed, and that list was constantly changing.

As a result, many of the packages are optional or completely unnecessary. I also added a few utilities to the install list which I felt might be helpful for troubleshooting.

The script was _never_ intended to be run on a real, long lived server. I would suggest using a Vagrant box (like I do, using the commented out commands at the top of the script), or a Docker/Podman container, either of which you can delete once the build process is complete.

That said, it was because of how much is involved in rebuildin these packages that I thought to upload the finished product, so others don't need to do the same.

**Comment by ladar** [comment]
> I have to update my 2 small Server to alma 9 or oracle 9 ??

I primarily use Alma these days. So that is what I am building/testing on. That said, these packages _should_ install and work perfectly on RHEL, Rocky, CentOS Stream and Oracle Linux, but I haven't tested on those variants. My breakup with CentOS was brutal, and painful, and I'm still recovering from the trauma, which has made me less receptive to new suitors.

I do have access to all of those distros though. So if you run into problems let me know. The only issue I can potentially forsee is a problem with the Unbreakable Enterprise Kernel (UEK) which Oracle Linux ships with. The OL repos do offer the unmodified normal kernel as well, so if that does become an issue, and you might try that instead.

**Comment by gueniewie** [comment]
Thanks for the clarification, I'm still thinking about what I can / should do, but I definitely need the virt-manager (usual program) and if I can get NEWER versions, then let's go...

Thanks again for your time and help!
Greetings from Salzburg AT
Guenther

**Comment by ladar** [comment]
So I just updated the scripts. And while I was doing so I realized that I made a mistake. I deleted the old package files, but didn't commit the deletions, like I intended to. As such the repo had two versions for some packaged. That is what caused he edk2 conflict. And it also made the download bigger. My apologies. Fix is committed, and pushed.

**Comment by gueniewie** [comment]
Hello Ladar ;-)
I actually wanted to "tell" you with my logs that there are two versions arguing about which one should be installed? I'm not familiar with git but I can "report" something. I'm 72 years old, but still interested in my Linux boxes.

I kept you on your toes, but we'll get there in time!

My BUILD process is still running but it's already on
# The spice-protocol rpmbuild finished. ;-)

**Comment by gueniewie** [comment]
I have a Problem with BUILD.sh look on the logs ? can I find any other Logs to correct my Problem?
I mean I have a Problem to Build this Packets I found nothing that this Packets are build?

# The spice rpmbuild finished.
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-common-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-docs-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-img-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-guest-agent-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-tools-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-pr-helper-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-virtiofsd-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-tests-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-block-blkio-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-block-curl-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-block-iscsi-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-block-rbd-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-block-ssh-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-ui-opengl-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-block-dmg-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-audio-alsa-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-audio-dbus-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-audio-oss-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-audio-pa-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-audio-sdl-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-ui-curses-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-ui-dbus-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-ui-gtk-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-ui-sdl-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-ui-egl-headless-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-char-baum-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-device-display-virtio-gpu-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-device-display-virtio-gpu-gl-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-device-display-virtio-gpu-pci-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-device-display-virtio-gpu-pci-gl-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-device-display-virtio-gpu-ccw-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-device-display-virtio-vga-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-device-display-virtio-vga-gl-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-device-usb-host-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-device-usb-redirect-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-device-usb-smartcard-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-device-display-vhost-user-gpu-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-ui-spice-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-ui-spice-app-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-device-display-qxl-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-char-spice-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-audio-spice-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-kvm-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-kvm-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-binfmt-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-static-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-static-aarch64-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-static-alpha-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-static-arm-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-static-cris-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-static-hexagon-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-static-hppa-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-static-loongarch64-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-static-m68k-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-static-microblaze-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-static-mips-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-static-nios2-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-static-or1k-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-static-ppc-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-static-riscv-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-static-s390x-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-static-sh4-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-static-sparc-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-static-x86-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-user-static-xtensa-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-aarch64-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-aarch64-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-alpha-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-alpha-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-arm-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-arm-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-avr-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-avr-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-cris-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-cris-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-hppa-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-hppa-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-loongarch64-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-loongarch64-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-m68k-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-m68k-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-microblaze-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-microblaze-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-mips-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-mips-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-nios2-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-nios2-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-or1k-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-or1k-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-ppc-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-ppc-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-riscv-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-riscv-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-rx-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-rx-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-s390x-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-s390x-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-sh4-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-sh4-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-sparc-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-sparc-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-tricore-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-tricore-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-x86-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-x86-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-xtensa-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-system-xtensa-core-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-debuginfo-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden
Fehler: Öffnen von /root/rpmbuild/RPMS/x86_64/qemu-debugsource-7.2.0-6.btrh9.x86_64.rpm fehlgeschlagen: Datei oder Verzeichnis nicht gefunden

# The qemu rpmbuild failed.

**Comment by ladar** [comment]
> I actually wanted to "tell" you with my logs that there are two versions arguing about which one should be installed? I'm not familiar with git but I can "report" something. I'm 72 years old, but still interested in my Linux boxes.

If that did happen, then doing a `git pull` to resync your environment will delete the duplicate files. And running the `INSTALL.sh` script again should fix the problem. If it doesn't can you open a new issue with the error message your seeing? You might have a package installed from the distro repos that isn't on the remove list.

**Comment by gueniewie** [comment]
Thanks for the clarification, I'm still thinking about what I can / should do, but I definitely need the virt-manager (usual program) and if I can get NEWER versions, then let's go..

Thanks again for your time and help!
Greetings from Salzburg AT
Guenther
